packages/block-library/src/navigation-submenu/edit.js

```diff
@@ -312,20 +315,39 @@ export default function NavigationSubmenuEdit( {
 		selectedBlockHasDescendants,
 		userCanCreatePages,
 		userCanCreatePosts,
+		onlyDescendantIsEmptyLink,
 	} = useSelect(
 		( select ) => {
 			const {
 				getClientIdsOfDescendants,
 				hasSelectedInnerBlock,
 				getSelectedBlockClientId,
 				getBlockParentsByBlockName,
+				getBlock,
 			} = select( blockEditorStore );
 
+			let _onlyDescendantIsEmptyLink;
+
 			const selectedBlockId = getSelectedBlockClientId();
 
 			const descendants = getClientIdsOfDescendants( [ clientId ] )
 				.length;
 
+			const selectedBlockDescendants = getClientIdsOfDescendants( [
+				selectedBlockId,
+			] );
+
+			// Check for a single descendant in the submenu. If that block
+			// is a link block in a "placeholder" state with no URL then
+			// we can consider as an "empty" link.
+			if ( selectedBlockDescendants?.length === 1 ) {
+				const singleBlock = getBlock( selectedBlockDescendants[ 0 ] );
+
+				_onlyDescendantIsEmptyLink =
+					singleBlock?.name === 'core/navigation-link' &&
```

Comment:
Nit: this is a very specific condition relating to a different block. I fear that a change in either the navigation-link logic or the default placeholder choice would break this in a hard to notice way. An E2E test should help here. By the way, is there a similar check anywhere else?

Reply:
Yes I don't love the tight coupling here.

Options:

1. Ask @javierarce how important he feels [this is to UX](https://github.com/WordPress/gutenberg/pull/38203#issuecomment-1027672948).
2. Add an e2e test to cover the scenario.

Reply:
Since we don't allow to revert submenus that contain items, the importance of that UI is probably limited and more of a helpful shortcut… but that said, I think is a nice shortcut because removing links is a bit tricky right now.

Reply:
@adamziel I've added e2e test coverage here.